main/modes/games/soko/soko.c

```diff
@@ -154,21 +155,22 @@ static void sokoMenuCb(const char* label, bool selected, uint32_t settingVal)
     if (selected)
     {
         // placeholder.
-        if (label == sokoResumeGameLabel)
-        {
-            int32_t data;
-            readNvs32("sk_data", &data);
-            // bitshift, etc, as needed.
-            uint16_t lastSaved = (uint16_t)data;
-            sokoLoadGameplay(soko, lastSaved, false);
-            sokoInitGameBin(soko);
-            soko->screen = SOKO_LEVELPLAY;
-        }
-        else if (label == sokoNewGameLabel)
+        if (label == sokoPlayGameLabel)
         {
-            // load level.
-            // we probably shouldn't have a new game option; just an overworld option.
-            sokoLoadGameplay(soko, 0, true);
+            int32_t overworld_player;
+            if(readNvs32("sk_overworldPos", &overworld_player))
+            {
+                soko->overworld_playerX = (uint16_t)(overworld_player & 0b1111111111111111);//okay so the cast to uint16 just does this right?
```

Comment:
Yes, though I like the mask for explicitness. `0xFFFF` is a bit more concise, but I counted the `1`s and you're good